File: backend/app/schemas/store.py
from pydantic import BaseModel

from app.models.enums import StoreCategory


class StoreListItem(BaseModel):
    id: int
    name: str
    category: StoreCategory
    description: str | None
    thumbnail: str | None
    phone: str | None
    owner_name: str | None


class StoreDetail(BaseModel):
    id: int
    market_id: int
    name: str
    category: StoreCategory
    description: str | None
    thumbnail: str | None
    phone: str | None
    owner_name: str | None
    discount_count: int

File: frontend/src/lib/api/store.ts
import { apiFetch } from "./client";

export type StoreListItem = {
  id: number;
  name: string;
  category: string;
  description: string | null;
  thumbnail: string | null;
  phone: string | null;
  owner_name: string | null;
};

export type StoreDetail = StoreListItem & {
  market_id: number;
  discount_count: number;
};

export function getStore(storeId: number): Promise<StoreDetail> {
  return apiFetch<StoreDetail>(`/stores/${storeId}`);
}
